Woof/Woof/View/OwnerProfileView/OwnerProfileView.swift

```diff
@@ -0,0 +1,55 @@
+import SwiftUI
+
+/// View which could be used to display or edit information about Owner
+struct OwnerProfileView: View {
+    /// View model responsible to manage data from model layer
+    @ObservedObject var viewModel = OwnerProfileViewModel()
+
+    /// Owner information editing status tracker
+    @State private var isEditingTracker = false
+
+    var body: some View {
+        VStack {
+            if isEditingTracker {
+                EditOwnerInformationView(name: $viewModel.name,
+                                         surname: $viewModel.surname,
+                                         phone: $viewModel.phone,
+                                         address: $viewModel.address)
+            } else {
+                OwnerCardView(name: viewModel.name,
+                              surname: viewModel.surname,
+                              phone: viewModel.phone,
+                              address: viewModel.address,
+                              avatarUrl: viewModel.avatarURL)
+            }
+            Spacer()
+        }
+        .padding(.horizontal)
+
+        .overlay(alignment: .topTrailing) {
+            Button(action: {
+                if !viewModel.name.isEmpty {
+                    viewModel.save()
+                    isEditingTracker.toggle()
+                }
+            }, label: {
+                Text(isEditingTracker ? "Save" : "Edit")
+                    .padding(.horizontal)
+                    .background(
+                        Capsule()
+                            .foregroundColor(!viewModel.name.isEmpty ? Color.App.purpleDark : Color.App.grayDark)
+                    )
+                    .foregroundColor(Color.App.white)
```

Comment:
nit:  ok for now
Think about creating a view modifier for button style, if we will use the dark purple/gray button with white text in our app somewhere else.

Reply:
ok, i'll create issue with this possible improvement